diff --git a/misc/zsh/completion.zsh b/misc/zsh/completion.zsh
index 8a899ec99..8fe877c22 100644
--- a/misc/zsh/completion.zsh
+++ b/misc/zsh/completion.zsh
@@ -22,7 +22,7 @@ function _nix() {
   elif [[ "$tpe" == attrs ]]; then
     args+=('-S' '')
   fi
-  compadd -J nix "${args[@]}" -d suggestions_display -a suggestions
+  compadd -Q -J nix "${args[@]}" -d suggestions_display -a suggestions
 }

 _nix "$@"
